src/Elastic.Apm/Metrics/MetricsProvider/BreakdownMetricsProvider.cs

```diff
@@ -91,12 +120,14 @@ public IEnumerable<MetricSet> GetSamples()
 			lock (_lock)
 			{
 				retVal = new List<MetricSet>(_itemsToSend.Count);
-				retVal.AddRange(_itemsToSend);
+				retVal.AddRange(_itemsToSend.Values.ToList());
 				_itemsToSend.Clear();
-				_transactionCount = 0;
 				_loggedWarning = false;
 			}
 
+			var timestampNow = TimeUtils.TimestampNow();
+			// According to the spec, timestampNow should be the time when we report the metrics.
+			foreach (var item in retVal) item.Timestamp = timestampNow;
```

Comment:
Since we need to touch each `item` anyway, we could replace the `AddRange` above with this loop and add each item individually to `retVal`.

Reply:
Thanks, good point! I change it.